nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/data/explorer/StandardKustoIngestService.java

```diff
@@ -0,0 +1,328 @@
+                    if (statuses1.get(0).status == OperationStatus.Succeeded
+                            || statuses1.get(0).status == OperationStatus.Failed
+                            || statuses1.get(0).status == OperationStatus.PartiallySucceeded) {
+                        future.complete(statuses1);
+                    }
+                } catch (Exception e) {
+                    future.completeExceptionally(new ProcessException("Error occurred while checking ingestion status", e));
+                }
+            };
+            scheduler.scheduleWithFixedDelay(task, 1, 2, TimeUnit.SECONDS);
+            statuses = future.get(1800, TimeUnit.SECONDS);
+            return KustoIngestionResult.fromString(statuses.get(0).status.toString());
+        } catch (IngestionClientException | IngestionServiceException e) {
+            throw new ProcessException("Error occurred while ingesting data into ADX", e);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            throw new ProcessException("Error occurred while checking ingestion status", e);
+        } finally {
+            shutDownScheduler(scheduler);
+        }
+    }
+
+    public void shutDownScheduler(ScheduledExecutorService executorService) {
+        executorService.shutdown();
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+                // Cancel currently executing tasks forcefully
+                executorService.shutdownNow();
+                // Wait a while for tasks to respond to being cancelled
+                if (!executorService.awaitTermination(60, TimeUnit.SECONDS))
+                    getLogger().error("Scheduler did not terminate");
+            }
+        } catch (InterruptedException ex) {
+            // (Re-)Cancel if current thread also interrupted
+            executorService.shutdownNow();
+        }
+    }
+
+    private ConnectionStringBuilder createKustoEngineConnectionString(final String clusterUrl,
+                                                                      final String appId,
+                                                                      final String appKey,
+                                                                      final String appTenant,
+                                                                      final KustoAuthenticationStrategy kustoAuthStrategy) {
+        final ConnectionStringBuilder builder;
+        if(KustoAuthenticationStrategy.APPLICATION_CREDENTIALS == kustoAuthStrategy){
+            builder = ConnectionStringBuilder.createWithAadApplicationCredentials(clusterUrl, appId, appKey, appTenant);
+        }else{
+            builder = ConnectionStringBuilder.createWithAadManagedIdentity(clusterUrl, appId);
+        }
+
+        final String vendor = System.getProperty("java.vendor");
+        final String version = System.getProperty("java.version");
```

Comment:
Is Kusto.Nifi.Sink better to distinguish source vs Sink ? Or have an additional pair Type->Sink ?

Reply:
Yeah Kusto.Nifi.Sink is a better identifier in the connector name although we have Pair<String, String> NIFI_SINK = Pair.of("processor", "nifi-sink"); as an additional field